app/nslc/management/commands/load_from_fdsn.py

```diff
@@ -0,0 +1,228 @@
+            f"&maxlat={params['maxlat']}"
+            f"&minlon={params['minlon']}"
+            f"&maxlon={params['maxlon']}"
+            f"&endafter={params['endafter']}"
+            "&format=text"
+        )
+        return url
+
+    def parse_datetime(self, datetime_str):
+        ''' parse datetime from string of form
+            # 2599-12-31T23:59:59
+            return datetime
+        '''
+        year, month, day_time = datetime_str.split("-")
+        day, time = day_time.split("T")
+        hour, minute, sec = time.split(":")
+        return datetime(int(year), int(month), int(day), int(hour),
+                        int(minute), int(sec[:2]), tzinfo=pytz.UTC)
+
+    '''Django command to check network and channel tables with FDSN service'''
+    def handle(self, *args, **options):
+        print('Getting data from FDSN...')
+        LOADER_EMAIL = os.environ.get('LOADER_EMAIL')
+        if not LOADER_EMAIL:
+            print(
+                "You must provide a valid email by setting the LOADER_EMAIL "
+                "environmental variable."
+            )
+            sys.exit(1)
+        project_path = options['path']
+        os.environ.setdefault('DJANGO_SETTINGS_MODULE', 'squac.settings')
+        sys.path.append(project_path)
+        django.setup()
+        try:
+            user = get_user_model().objects.get(email=LOADER_EMAIL)
+        except ObjectDoesNotExist:
+            print(
+                f"Loader email {LOADER_EMAIL} does not exist.\n"
+                "You must provide a valid email by setting the LOADER_EMAIL "
+                "environmental variable."
+            )
+            sys.exit(1)
+
+        # create hash for lookup
+        networks = {}
+        for n in Network.objects.all():
+            networks[n.code.lower()] = n
+
+        station_url = self.build_url(options, 'station')
+        # create a station hash so we can pull out station description
+        stations = {}
+        with requests.Session() as s:
+            download = s.get(station_url)
+            decoded_content = download.content.decode('utf-8')
+            content = csv.reader(decoded_content.splitlines(), delimiter='|')
+            row_list = list(content)
+            # skip first two rows of metadata
+            for row in row_list[3:-2]:
+                if len(row) == 8:
+                    sta_code = row[1].lower()
+                    sta_name = row[5]
+                    stations[sta_code] = sta_name
+
+        channel_url = self.build_url(options, 'channel')
+        with requests.Session() as s:
+            download = s.get(channel_url)
+            decoded_content = download.content.decode('utf-8')
+            content = csv.reader(decoded_content.splitlines(), delimiter='|')
+            row_list = list(content)
+            # skip header and empty rows in metadata
+            for row in row_list[3:-2]:
+                # extract data from row
+                if len(row) == 17:
```

Comment:
Will this row always have a length of 17?

Reply:
This now checks for rows of length greater than one as comments should be an array of 1 element.